src/Tabs.tsx

```diff
@@ -1 +1,35 @@
-// implement a component here
+import React, { useEffect } from 'react';
+
+type Props = {
+  tabs: Tab[],
+  selectedTabId: string,
+  onTabSelected: (tab: Tab) => void
+};
+
+// eslint-disable-next-line max-len
+export const Tabs:React.FC<Props> = ({ tabs, selectedTabId, onTabSelected }) => {
+  let selectedTab = tabs.find(tab => tab.id === selectedTabId);
+
+  useEffect(() => {
+    selectedTab = tabs.find(tab => tab.id === selectedTabId);
+  }, [selectedTabId]);
```

Comment:
It's redundant, delete it
When this component receives new props, all this code start to run, and on 11 line selectedTab variable change it's value
so useEffect is redundant here